hircluster.c

```diff
@@ -1189,6 +1193,89 @@ dict *parse_cluster_nodes(redisClusterContext *cc, char *str, int str_len,
     return NULL;
 }
 
+/* Update known cluster nodes with a new collection of redisClusterNodes.
+ * Will also update the slot-to-node lookup table for the new nodes. */
+static int updateNodesAndSlotmap(redisClusterContext *cc, dict *nodes) {
+    if (nodes == NULL) {
+        return REDIS_ERR;
+    }
+
+    /* Create a slot to redisClusterNode lookup table */
+    redisClusterNode **table;
```

Comment:
If this function is placed after cluster_update_route_by_addr() it would be a smaller diff. The bulk of this code would not have been moved.